### PR TITLE
26815 multisite keyword redirect

### PR DESCRIPTION
**Pull request by igor-dovgaliuk**
Make Orckestra.Search.KeywordRedirect work per website
AB#26815

### CODE REVIEW

File: Orckestra.Search.KeywordRedirect/Orckestra.Search.KeywordRedirect.Package/Package/install.xml [hunk 1]
```diff
@@ -1,24 +1,25 @@
...
-				<File sourceFilename="~\App_Data\Composite\TreeDefinitions\KeywordRedirect.xml" targetFilename="~\App_Data\Composite\TreeDefinitions\KeywordRedirect.xml" allowOverwrite="false" />
-				<File sourceFilename="~\Bin\Orckestra.Search.KeywordRedirect.dll" targetFilename="~\Bin\Orckestra.Search.KeywordRedirect.dll" allowOverwrite="false" />
-			</Files>
-			<Directories>
-				<Directory sourceDirectory="~\Composite\InstalledPackages\Orckestra.Search.KeywordRedirect\" targetDirectory="~\Composite\InstalledPackages\Orckestra.Search.KeywordRedirect\" allowOverwrite="false" deleteTargetDirectory="false" />
-			</Directories>
-		</mi:Add>
-	</mi:PackageFragmentInstallers>
+  <mi:PackageRequirements minimumCompositeVersion="6.2.6394.36351" maximumCompositeVersion="9.9999.9999.9999" />
+  <mi:PackageInformation name="Orckestra.Search.KeywordRedirect" groupName="Orckestra.Search" version="1.0.2" author="Orckestra Technologies" website="https://c1.orckestra.com/" readMoreUrl="https://docs.c1.orckestra.com/Orckestra.Search.KeywordRedirect" id="3fffb96f-2447-4056-a5a0-db1cef901d23" canBeUninstalled="true" systemLocking="hard" flushOnCompletion="true" allowOverwrite="false">
```
Comment (InnaBoitsun) [inline Orckestra.Search.KeywordRedirect/Orckestra.Search.KeywordRedirect.Package/Package/install.xml]: When we create version of package or fix some bug for package we need to specify new version in install.xml. Please specify version="**1.0.3**" Also when you will megre branch to master, start the comment with [PackageName] - version - description/ Example: 
[Orckestra.Search.KeywordRedirect] 1.0.3 - support multi websites

Comment (igor-dovgaliuk) [reply Orckestra.Search.KeywordRedirect/Orckestra.Search.KeywordRedirect.Package/Package/install.xml]: Done, thanks

File: Orckestra.Search.KeywordRedirect/Orckestra.Search.KeywordRedirect/KeywordManager.cs [hunk 2]
```diff
@@ -41,86 +50,146 @@ public void OnError(Exception error)
...
+
+                string GetPageUrl(RedirectKeyword redirectKeyword)
+                {
+                    return PageUrls.BuildUrl(new PageUrlData(redirectKeyword.LandingPage, PublicationScope.Published, cultureInfo));
+                }
+            }
+        }
 
-            if (!keywords.TryGetValue(keyword, out string result))
+        private void FixMissingHomePages()
```
[the remark this answers is not in the document]
Comment (igor-dovgaliuk) [reply Orckestra.Search.KeywordRedirect/Orckestra.Search.KeywordRedirect/KeywordManager.cs]: No,  is doesn't change publication status
How I've tested:
1. Added keywords with old package
2. Published some of them
3. Uninstalled package & installed new version
=> Homepage is set and status preserved

File: Orckestra.Search.KeywordRedirect/Orckestra.Search.KeywordRedirect/KeywordManager.cs [hunk 3]
```diff
@@ -41,86 +50,146 @@ public void OnError(Exception error)
...
             }
+
             return result;
         }
 
-        public string GetPublicRedirect(string keyword, CultureInfo cultureInfo)
+        private Guid GetHomePageIdByPageId(Guid pageId)
         {
-            if (keyword == null)
+            return _pageIdToHomePageIdCache.GetOrAdd(pageId, GetHomePageId);
```
[the remark this answers is not in the document]
Comment (igor-dovgaliuk) [reply Orckestra.Search.KeywordRedirect/Orckestra.Search.KeywordRedirect/KeywordManager.cs]: Added cache clean, thanks

**Pull request merged**